### PR TITLE
open-api: add v3; add spec from http

### LINKED ISSUE #423: openApiSpecUrl does not work for HTTP URL
When specifying an HTTP URL (e.g. 'http://localhost:8080/v2/api-docs', which is the URL for the API docs in the demo app in webtau-testapp) for the openApiSpecUrl parameter in a webtau.properties or webtau.cfg file, the test fails because it cannot find the spec:
```
Caused by: java.lang.IllegalArgumentException: Unable to load API descriptor from provided "http:/localhost:8080/v2/api-docs":
	[unable to read location `"http:/localhost:8080/v2/api-docs"`]
	at com.twosigma.webtau.openapi.OpenApiSpec.<init>(OpenApiSpec.java:56)
	at com.twosigma.webtau.openapi.OpenApi.reset(OpenApi.java:49)
	at com.twosigma.webtau.openapi.OpenApiSpecConfig.onAfterCreate(OpenApiSpecConfig.java:45)
	at com.twosigma.webtau.cfg.WebTauConfig.lambda$triggerConfigHandlers$16(WebTauConfig.java:116)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at com.twosigma.webtau.cfg.WebTauConfig.triggerConfigHandlers(WebTauConfig.java:116)
	at com.twosigma.webtau.cfg.WebTauConfig.<init>(WebTauConfig.java:104)
	at com.twosigma.webtau.cfg.WebTauConfig$CfgInstanceHolder.<clinit>(WebTauConfig.java:376)
	... 31 more
```
You can see in the error message that the double slash in the URL is being converted to a single slash. It looks like this is caused by the OpenApiSpecConfig.java's onAfterCreate method.

### PR DESCRIPTION
**Pull request by MykolaGolubyev**
closes #423

### CODE REVIEW

File: webtau-open-api/src/main/java/org/testingisdocumenting/webtau/openapi/OpenApiSpec.java [hunk 1]
```diff
@@ -79,44 +67,36 @@ public String getSpecUrl() {
...
-        return Optional.of(new OpenApiOperation(method,
-                combineWithBasePath(apiOperation.getApiOperation().getApiPath().original())));
+        return Optional.of(new OpenApiOperation(method, apiOperation.getApiOperation().getApiPath().original()));
     }
 
     public Set<String> getDeclaredResponses(OpenApiOperation op) {
         return operationsAndResponses.getOrDefault(op, Collections.emptySet());
     }
 
-    private String combineWithBasePath(String url) {
```
Comment (MykolaGolubyev) [inline webtau-open-api/src/main/java/org/testingisdocumenting/webtau/openapi/OpenApiSpec.java:94]: @tsiq-karold I haven't found a test that relies on it, wonder if it was somewhere on your side. Also couldn't find an alternative to the base path in new API.

File: webtau-open-api/src/test/groovy/org/testingisdocumenting/webtau/openapi/OpenApiResponseValidatorTest.groovy [hunk 2]
```diff
@@ -133,6 +133,7 @@ class OpenApiResponseValidatorTest implements StepReporter {
     private static HttpResponse response() {
         def response = new HttpResponse()
         response.setTextContent('{"key": "value"}')
+        response.setContentType('application/json')
```
Comment (MykolaGolubyev) [inline webtau-open-api/src/test/groovy/org/testingisdocumenting/webtau/openapi/OpenApiResponseValidatorTest.groovy:136]: that took me an hour to figure out :(

File: pom.xml [hunk 3]
```diff
@@ -514,6 +507,60 @@
...
+            <dependency>
+                <groupId>com.fasterxml.jackson.dataformat</groupId>
+                <artifactId>jackson-dataformat-yaml</artifactId>
+                <version>${jackson.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>io.swagger</groupId>
+                <artifactId>swagger-core</artifactId>
+                <version>1.6.1</version>
```
Comment (tsiq-karold) [inline pom.xml:519]: Why not `swagger.version`?

File: pom.xml [hunk 4]
```diff
@@ -514,6 +507,60 @@
...
+                    <exclusion>
+                        <groupId>javax.validation</groupId>
+                        <artifactId>validation-api</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
+
+            <dependency>
+                <groupId>com.sun.xml.bind</groupId>
+                <artifactId>jaxb-impl</artifactId>
```
Comment (tsiq-karold) [inline pom.xml:554]: Oh boy, why do we need this?

File: webtau-open-api/src/main/java/org/testingisdocumenting/webtau/openapi/OpenApiSpec.java (hunk 1 above)
Comment (tsiq-karold) [reply webtau-open-api/src/main/java/org/testingisdocumenting/webtau/openapi/OpenApiSpec.java:94]: I vaguely recall writing this but I cannot remember the specifics.  I don't think I need it.

File: webtau-open-api/src/test/groovy/org/testingisdocumenting/webtau/openapi/OpenApiResponseValidatorTest.groovy (hunk 2 above)
Comment (tsiq-karold) [reply webtau-open-api/src/test/groovy/org/testingisdocumenting/webtau/openapi/OpenApiResponseValidatorTest.groovy:136]: :(

File: pom.xml (hunk 3 above)
Comment (MykolaGolubyev) [reply pom.xml:519]: Yeah, I removed it and put it back fighting enforcer. Maybe I don't need it now. 
I will replace with version if it is still needed. 
On the other hand, versions through properties prevent maven auto update versions via command line, but that's not a good enough reason

File: pom.xml (hunk 4 above)
Comment (MykolaGolubyev) [reply pom.xml:554]: me fighting with dependencies (note, the exclusion bit), may need to try to remove it again.

**Review by tsiq-karold — APPROVED** [review]

**Pull request merged**